client/src/js/services/TrialBalanceService.js

```diff
@@ -163,5 +147,28 @@ function JournalPostingModalService(util, $http, Modal) {
     return $http.post(url, {transactions : transactions});
   }
 
+  /**
+   * @function getRelatedTransaction
+   * @description
+   * This function gets an account_id with a list of transaction and send back
+   * a list of transaction related to this account
+   **/
+  function getRelatedTransaction(accountId, records){
+    var transactions = [], processed = [];
+
+    records.forEach(function (item) {
+      if(processed.indexOf(item.trans_id) === -1 && item.account_id === accountId){
+        transactions = transactions.concat(getTransaction(item.trans_id, records));
+      }
+    });
+    return transactions;
+  }
+
+  function getTransaction(transId, records){
```

Comment:
Of course.  But it is better than `O(2n)`, which is also `O(n)`.  I abused the notation to try and be precise